core/app/models/spree/store.rb

```diff
@@ -47,9 +47,12 @@ class Store < Spree::Base
 
     alias_attribute :contact_email, :customer_support_email
 
-    def self.current(domain = nil)
-      current_store = domain ? Store.by_url(domain).first : nil
-      current_store || Store.default
+    def self.current(url = nil)
+      ActiveSupport::Deprecation.warn(<<-DEPRECATION, caller)
```

Comment:
Shouldn't this warning be added to the `default` method?

Reply:
@mad-eel nope, we only need to deprecate the `current` method as we've moved this logic to the finder class which can be easily swapped for a custom one by developers